internal/querynode/segment_loader.go

```diff
@@ -752,8 +784,14 @@ func (loader *segmentLoader) checkSegmentSize(collectionID UniqueID, segmentLoad
 						zap.Int64("indexBuildID", fieldIndexInfo.BuildID))
 					return err
 				}
-				usedMemAfterLoad += neededMemSize
 				usedLocalSizeAfterLoad += neededDiskSize
+				// diskann not need to copy data
+				indexType, _ := funcutil.GetAttrByKeyFromRepeatedKV("index_type", fieldIndexInfo.IndexParams)
+				if indexType == indexparamcheck.IndexDISKANN {
+					usedMemAfterLoad += uint64(float64(neededMemSize) / Params.QueryNodeCfg.LoadMemoryUsageFactor)
```

Comment:
I think this is a bug
I guess we thought `memLoadingUsage = usedMemAfterLoad / LoadMemoryUsageFactor`, and for DiskANN we don't need the loading expansion factors. So we div LoadMemoryUsageFactor as a preprocessing.
But `memLoadingUsage = usedMemAfterLoad + maxSegmentSize * concurrency * LoadMemoryUsageFactor`, so we should preprocess the second part instead of the first.